### ISSUE TITLE
Improve error messages from unfulfilled expectations

### ISSUE THREAD

**Issue by MarkKoester** [rollbear/trompeloeil#153]
Currently, the error messages that are generated from unfulfilled expectations are cryptic and hard to debug. I almost always have to step through my failing test with a debugger to figure out what the issue is.

Improving these error messages would definitely make this library more user-friendly.

**Comment by rollbear** [comment]
OK, can you give some hints about information you are missing that you'd like to see, or example of information that is misleading that can be improved?

**Comment by MarkKoester** [comment]
Here's one example:
<img width="634" alt="Screen Shot 2019-08-02 at 2 28 54 PM" src="https://user-images.githubusercontent.com/24963825/62391291-69d3ef00-b532-11e9-8ad1-c8b416358dde.png">

First, there's a lot of unnecessary information in this message. I feel like most of text here isn't all that relevant when debugging besides "No match for call to receiveAck...", so most of it is noise besides a portion of the middle line.

Second, it's hard to parse exactly what caused the error from the message. I can tell that receiveAck was called when it shouldn't have been, but that's basically all I know. I still have to do some extra work to answer these questions:
- Did the test case expect receiveAck to never be called and it was called once?
- Did the test case expect a call to receiveAck but the argument didn't match?
- Did the test case expect receiveAck to be called 3 times and it was actually called 4 times?

Basically, I think it would be awesome if the error messages were less noisy and more targeted, and clearly highlighted the difference between the expected behavior and the actual behavior.

Thanks for taking the time to respond to this.
